llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -165,33 +170,109 @@ void InterBlockScheduling::leaveFunction() {
 }
 
 void InterBlockScheduling::enterBlock(MachineBasicBlock *BB) {
-  CurrentBlock = &getBlockState(BB);
-  CurrentBlock->resetRegion();
+  CurrentBlockState = &getBlockState(BB);
+  CurrentBlockState->resetRegion();
   DEBUG_BLOCKS(dbgs() << "  >> enterBlock " << BB->getNumber() << " "
-                      << CurrentBlock->kindAsString() << " FixPointIter="
-                      << CurrentBlock->FixPoint.NumIters << "\n");
-}
+                      << CurrentBlockState->kindAsString() << " FixPointIter="
+                      << CurrentBlockState->FixPoint.NumIters
+                      << " II=" << CurrentBlockState->FixPoint.II);
+}
+namespace {
```

Comment:
As this file is growing, maybe we can move this class to AIEPostPipeliner* files.

Reply:
Perhaps not the best criterion to do this. It implements an interface to the pipeliner that is dedicated to the interblock data structures.  All that knowledge would need to be exported to the postpipeliner, which would clutter it.